src/twisted/application/runner/test/test_runner.py

```diff
@@ -68,11 +68,31 @@ def test_runInOrder(self):
         """
         L{Runner.run} calls the expected methods in order.
         """
-        runner = DummyRunner({})
+
+        calledMethods = []
```

Comment:
My general feeling is normally that it's not worth doing this kind of detangling, especially if it's a single place for testing purposes (I'm typically opposed to the idea of "interface-for-testing" as a philosophy).

That is, I suspect (without having enough understanding of what I'm seeing in this PR) that it's not worth doing the cleanup. I'd love to look at the notion in more detail at PyCon though if you're interested - it could be an informative exploration.